packages/frontend/editor-ui/src/stores/builder.store.ts

```diff
@@ -0,0 +1,361 @@
+import { chatWithBuilder } from '@/api/ai';
+import type { VIEWS } from '@/constants';
+import { EDITABLE_CANVAS_VIEWS, STORES, WORKFLOW_BUILDER_EXPERIMENT } from '@/constants';
+import type { ChatRequest } from '@/types/assistant.types';
+import type { ChatUI } from '@n8n/design-system/types/assistant';
+import { defineStore } from 'pinia';
+import { computed, ref, watch } from 'vue';
+import { useRootStore } from './root.store';
+import { useUsersStore } from './users.store';
+import { useRoute } from 'vue-router';
+import { useSettingsStore } from './settings.store';
+import { assert } from '@n8n/utils/assert';
+import { useI18n } from '@/composables/useI18n';
+import { useTelemetry } from '@/composables/useTelemetry';
+import { useUIStore } from './ui.store';
+import { usePostHog } from './posthog.store';
+import { useNodeTypesStore } from './nodeTypes.store';
+
+export const MAX_CHAT_WIDTH = 425;
+export const MIN_CHAT_WIDTH = 250;
+export const DEFAULT_CHAT_WIDTH = 330;
+export const ENABLED_VIEWS = [...EDITABLE_CANVAS_VIEWS];
+
+export const useBuilderStore = defineStore(STORES.BUILDER, () => {
+	// Core state
+	const chatWidth = ref<number>(DEFAULT_CHAT_WIDTH);
+	const chatMessages = ref<ChatUI.AssistantMessage[]>([]);
+	const chatWindowOpen = ref<boolean>(false);
+	const streaming = ref<boolean>(false);
+	const currentSessionId = ref<string | undefined>();
+	const assistantThinkingMessage = ref<string | undefined>();
+
+	// Store dependencies
+	const settings = useSettingsStore();
+	const rootStore = useRootStore();
+	const usersStore = useUsersStore();
+	const uiStore = useUIStore();
+	const route = useRoute();
+	const locale = useI18n();
+	const telemetry = useTelemetry();
+	const posthogStore = usePostHog();
+	const nodeTypesStore = useNodeTypesStore();
+
+	// Computed properties
+	const isAssistantEnabled = computed(() => settings.isAiAssistantEnabled);
+
+	const workflowPrompt = computed(() => {
+		const firstUserMessage = chatMessages.value.find(
+			(msg) => msg.role === 'user' && msg.type === 'text',
+		) as ChatUI.TextMessage;
+
+		return firstUserMessage?.content;
+	});
+	const canShowAssistant = computed(
+		() => isAssistantEnabled.value && ENABLED_VIEWS.includes(route.name as VIEWS),
+	);
+
+	const canShowAssistantButtonsOnCanvas = computed(
+		() => isAssistantEnabled.value && EDITABLE_CANVAS_VIEWS.includes(route.name as VIEWS),
+	);
+
+	const isAssistantOpen = computed(() => canShowAssistant.value && chatWindowOpen.value);
+
+	const isAIBuilderEnabled = computed(() => {
+		return (
+			posthogStore.getVariant(WORKFLOW_BUILDER_EXPERIMENT.name) ===
+			WORKFLOW_BUILDER_EXPERIMENT.variant
+		);
+	});
+
+	// No need to track unread messages in the AI Builder
+	const unreadCount = computed(() => 0);
```

Comment:
Nothing serious, but it would be nice if we can show popup when response is received while sidebar is closed (especially if something went wrong during generation and you don't get a workflow in the canvas):
![image](https://github.com/user-attachments/assets/38f18e0a-9b8c-4a59-9661-6ea0b87dc4cb)